ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlStatementException.java

```diff
@@ -17,34 +17,112 @@
 
 public class KsqlStatementException extends KsqlException {
 
+  public enum Problem {
```

Comment:
KsqlStatementException has become hideous. It's basically handling all the complexity of making sure that query fragments can still flow back to the API response layer without actually being reported in the message or stacktrace of any exceptions that might get logged.

I think we could clean it up substantially by changing up the whole ksql exception hierarchy, but I'd like to treat that as a separate effort. For now, I'm focusing on not logging query text while also not breaking the API.

This has already been enough of a challenge that I don't want to drag in a more extensive refactor in the same PR.